### PR TITLE
#37 redirect to the elimu.ai Appstore's GitHub page

### PR DESCRIPTION
**Pull request by gscdev**

### CODE REVIEW

File: app/src/main/java/ai/elimu/launcher/MainActivity.java [hunk 1]
```diff
@@ -21,18 +24,25 @@ protected void onCreate(Bundle savedInstanceState) {
...
+
         } catch (PackageManager.NameNotFoundException e) {
             Timber.w( null, e);
             Toast.makeText(getApplicationContext(), "The elimu.ai Launcher will not work without the elimu.ai Appstore: " + BuildConfig.APPSTORE_APPLICATION_ID, Toast.LENGTH_LONG).show();
-            // TODO: Add link to GitHub for downloading APK
-        }
 
-        Intent intent = new Intent(getApplicationContext(), HomeScreensActivity.class);
-        startActivity(intent);
+            goToAppstoreDownload();
```
Comment (nya-elimuai) [inline app/src/main/java/ai/elimu/launcher/MainActivity.java:35]: @gscdev It's a bit confusing to be redirected to a URL without any form of warning about what's happening. Maybe we could use a [snackbar](https://material.io/develop/android/components/snackbars) (or dialog) with an action button for this?

**Review by nya-elimuai — CHANGES_REQUESTED** [review]
I suggest adding more information for the user to understand that he/she will be redirected to a web browser.

File: app/src/main/java/ai/elimu/launcher/MainActivity.java (hunk 1 above)
Comment (gscdev) [reply app/src/main/java/ai/elimu/launcher/MainActivity.java:35]: Something like this?

Screenshot from 2020-09-22 01-13-12

File: app/src/main/java/ai/elimu/launcher/MainActivity.java [hunk 2]
```diff
@@ -17,18 +14,8 @@
...
-        } catch (PackageManager.NameNotFoundException e) {
-            Timber.w( null, e);
-            Toast.makeText(getApplicationContext(), "The elimu.ai Launcher will not work without the elimu.ai Appstore: " + BuildConfig.APPSTORE_APPLICATION_ID, Toast.LENGTH_LONG).show();
-            // TODO: Add link to GitHub for downloading APK
-        }
-
-        Intent intent = new Intent(getApplicationContext(), HomeScreensActivity.class);
-        startActivity(intent);
+            Intent intent = new Intent(getApplicationContext(), HomeScreensActivity.class);
+            startActivity(intent);
```
Comment (nya-elimuai) [inline app/src/main/java/ai/elimu/launcher/MainActivity.java:18]:
```suggestion
        Intent intent = new Intent(getApplicationContext(), HomeScreensActivity.class);
        startActivity(intent);
```

File: app/src/main/java/ai/elimu/launcher/ui/HomeScreensActivity.java [hunk 3]
```diff
@@ -152,6 +158,30 @@ public void onPageScrollStateChanged(int state) {
...
+                        public void onClick(@NonNull MaterialDialog dialog, @NonNull DialogAction which) {
+                            goToAppstoreDownload();
+                        }
+                    })
+                    .show();
+        }
+    }
+
+    private void goToAppstoreDownload() {
+        Intent intent = new Intent(Intent.ACTION_VIEW, Uri.parse(APPSTORE_DOWNLOAD_URL));
```
Comment (nya-elimuai) [inline app/src/main/java/ai/elimu/launcher/ui/HomeScreensActivity.java:181]: Maybe move this code into `onClick` (line 173), to reduce the number of methods?

**Review by nya-elimuai — APPROVED** [review]

File: app/src/main/java/ai/elimu/launcher/MainActivity.java (hunk 1 above)
Comment (nya-elimuai) [reply app/src/main/java/ai/elimu/launcher/MainActivity.java]: @gscdev Yes, perfect :-)

**Review by jo-elimuai — DISMISSED** [review]

**Review by nya-elimuai — APPROVED** [review]

**Review by jo-elimuai — APPROVED** [review]

**Pull request merged**